rust/noosphere-fs/src/fs.rs

```diff
@@ -276,13 +276,17 @@ where
         additional_headers: Option<Vec<(String, String)>>,
     ) -> Result<Cid> {
         let sphere = Sphere::at(&self.sphere_revision, &self.db);
-
-        // TODO(#85): Error result if mutation is empty and also additional headers are empty
-
-        let mut revision = sphere.try_apply_mutation(self.require_mutation()?).await?;
+        let mutation = self.require_mutation()?;
+        let mut revision = sphere.try_apply_mutation(mutation).await?;
 
         if let Some(mut headers) = additional_headers {
-            revision.memo.headers.append(&mut headers);
+            if headers.len() > 0 {
+                revision.memo.headers.append(&mut headers);
+            } else if mutation.is_empty() {
```

Comment:
Maybe we could try a `match` to clean up the conditions a little? Something like...

```rs
match additional_headers {
  Some(mut headers) if headers.len() > 0 => {
    // Append
  },
  _ if mutation.is_empty() => {
    // Error
  }
}
```

Reply:
If interested, please feel free to clean this up in a small PR. Otherwise I'm sure I can get to it eventually. Thanks again!